### PR TITLE
Add API to configure severity of compiler option to report unused lambda

### PR DESCRIPTION
**Pull request by noopur2507**
parameter

Fixes https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3102

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by noopur2507** [comment]
@srikanth-sankaran can you please review this PR?

**Comment by srikanth-sankaran** [comment]
> @srikanth-sankaran can you please review this PR?

yes, later today, I will.

File: org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java [hunk 1]
```diff
@@ -454,6 +454,19 @@ public final class JavaCore extends Plugin {
 	 * @category CompilerOptionID
 	 */
 	public static final String COMPILER_PB_UNUSED_LOCAL = PLUGIN_ID + ".compiler.problem.unusedLocal"; //$NON-NLS-1$
+	/**
+	 * Compiler option ID: Reporting Unused Lambda Parameter.
+	 * <p>When enabled, the compiler will issue an error or a warning for unused lambda
+	 *    parameter (that is, lambda parameter never read from).</p>
+	 * <dl>
```
Comment (srikanth-sankaran) [inline org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java:461]: Suggest use plural i.e for unused lambda parameters (that is lambda parameters never read from)

**Review by srikanth-sankaran — APPROVED** [review]
Minor editorial comment, looks good otherwise, Thanks Noopur!

File: org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java (hunk 1 above)
Comment (noopur2507) [reply org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java:461]: Updated.

**Comment by noopur2507** [comment]
> Minor editorial comment, looks good otherwise, Thanks Noopur!

Thanks, Srikanth.

**Pull request merged**